frontend/lib/src/components/widgets/DataFrame/hooks/useDataExporter.ts

```diff
@@ -100,40 +142,67 @@ function useDataExporter(
       const nativeFileSystemAdapter = await import(
         "native-file-system-adapter"
       )
-
-      const timestamp = new Date().toISOString().slice(0, 16).replace(":", "-")
-      const suggestedName = `${timestamp}_export.csv`
-
       const fileHandle = await nativeFileSystemAdapter.showSaveFilePicker({
         suggestedName,
         types: [{ accept: { "text/csv": [".csv"] } }],
         excludeAcceptAllOption: false,
       })
 
-      const textEncoder = new TextEncoder()
       const writer = await fileHandle.createWritable()
 
-      // Write UTF-8 BOM for excel compatibility:
-      await writer.write(textEncoder.encode(CSV_UTF8_BOM))
+      await writeCsv(writer, getCellContent, columns, numRows)
+    } catch (error) {
+      if (error instanceof Error && error.name === "AbortError") {
+        // The user has canceled the save dialog. Do nothing.
+        return
+      }
 
-      // Write headers:
-      const headers: string[] = columns.map(column => column.name)
-      await writer.write(textEncoder.encode(toCsvRow(headers)))
+      try {
+        logWarning(
+          "Failed to export data as CSV with FileSystem API, trying fallback method",
+          error
+        )
+        // Simulated WritableStream that builds CSV content in-memory for the Blob fallback method
+        let csvContent = ""
+
+        const inMemoryWriter = new WritableStream({
+          write: async chunk => {
+            csvContent += new TextDecoder("utf-8").decode(chunk)
+          },
+          close: async () => {},
+        })
+
+        await writeCsv(
+          inMemoryWriter.getWriter(),
+          getCellContent,
+          columns,
+          numRows
+        )
 
-      for (let row = 0; row < numRows; row++) {
-        const rowData: any[] = []
-        columns.forEach((column: BaseColumn, col: number, _map) => {
-          rowData.push(column.getCellValue(getCellContent([col, row])))
+        // Fallback to the old browser download method:
+        const blob = new Blob([csvContent], {
+          type: "text/csv;charset=utf-8;",
         })
-        // Write row to CSV:
-        await writer.write(textEncoder.encode(toCsvRow(rowData)))
+        const url = URL.createObjectURL(blob)
+        const link = document.createElement("a")
+        if (enforceDownloadInNewTab) {
+          link.setAttribute("target", "_blank")
```

Comment:
nit: maybe add a one-liner comment here why we are doing this

Reply:
I meant more like answering why we are doing that: "Open the download link in a new tab to ensure that this is working in embedded setups that limit the URL that an iframe can navigate to (e.g. via CSP)."

Reply:
I rephrased  with your suggestion!